lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -2187,6 +2217,32 @@ int s1ap_mme_handle_handover_request(
         ho_request_p->e_rab_list.item[i]
             .e_rab_level_qos_parameters.allocation_and_retention_priority
             .pre_emption_vulnerability;
+
+    // data forwarding not supported
+    OAILOG_INFO(LOG_S1AP, "Note: data forwarding unsupported\n");
+    S1ap_E_RABToBeSetupItemHOReq_ExtIEs_t* exts =
+        (S1ap_E_RABToBeSetupItemHOReq_ExtIEs_t*) calloc(
+            1, sizeof(S1ap_E_RABToBeSetupItemHOReq_ExtIEs_t));
+    exts->id          = S1ap_ProtocolIE_ID_id_Data_Forwarding_Not_Possible;
+    exts->criticality = S1ap_Criticality_ignore;
+    exts->extensionValue.present =
+        S1ap_E_RABToBeSetupItemHOReq_ExtIEs__extensionValue_PR_Data_Forwarding_Not_Possible;
+    exts->extensionValue.choice.Data_Forwarding_Not_Possible =
+        S1ap_Data_Forwarding_Not_Possible_data_Forwarding_not_Possible;
+
+    S1ap_ProtocolExtensionContainer_7327P1_t* xc =
+        (S1ap_ProtocolExtensionContainer_7327P1_t*) calloc(
+            1, sizeof(S1ap_ProtocolExtensionContainer_7327P1_t));
+    int asn_ret = 0;
+    asn_ret     = ASN_SEQUENCE_ADD(&xc->list, exts);
+    if (asn_ret) {
+      OAILOG_ERROR(LOG_S1AP, "ASN_SEQUENCE_ADD ret = %d\n", asn_ret);
+    }
+
+    // Bad cast...
+    e_RABToBeSetup->iE_Extensions =
```

Comment:
I see a forward declaration of `struct S1ap_ProtocolExtensionContainer` but do not see an actual definition. how does this work?

Reply:
Defined in generated s1ap structs by asn1 tool

Reply:
I'm not seeing the actual definition in that generated header.

Reply:
precisely what I meant :)